### PR TITLE
adding go teal compile option changes

### PR DESCRIPTION
**Pull request by barnjamin**
This is done in order to pass a GET parameter through the Go client for TEAL compile endpoint

### CODE REVIEW

**Review by Eric-Warehime — APPROVED** [review]

**Review by michaeldiamant — APPROVED** [review]
@barnjamin Thanks for finding a way forward here.  ☕ 

* I confirmed the generated changes produce _no_ diff with https://github.com/algorand/go-algorand-sdk/pull/335 as of https://github.com/algorand/go-algorand-sdk/pull/335/commits/2c673b256b64f3bd84058c94b48bc8f28d873c64.
* Out of scope to the PR:  I opened an issue because it feels to me that a more robust solution is warranted - https://github.com/algorand/generator/issues/38.

**Review by winder — CHANGES_REQUESTED** [review]
We shouldn't add special cases in new endpoints. The special cases elsewhere were done for backwards compatibility and will be removed in a future major release.

File: go_templates/query.vm [hunk 1]
```diff
@@ -40,6 +40,9 @@ import (
 #if ( $queryType == "PendingTransactionsByAddress" || $queryType == "PendingTransactionInformationByAddress" || $queryType == "PendingTransactionInformation" || $queryType == "PendingTransactions" || $queryType == "Block" || $queryType == "SuggestedParams" )
   "github.com/algorand/go-algorand-sdk/types"
 #end
+#if ( $queryType == "TealCompile")
```
[the remark this answers is not in the document]
Comment (winder) [reply go_templates/query.vm:43]: You need something like
```#if ( $q.method == "post" )
#set( $verb = "post" )
#else if (  $q.method == "post" )
#set( $verb = "get" )
#else
UNHANDLED HTTP METHOD: $q.method
##$unknown.type ## force a template failure with an unknown type
#end
```

Then you can use $verb when calling the helper function.


For the dependency, wrap it in a local helper like we do with post/get. That you don't can avoid adding a conditional import.

[the remark this answers is not in the document]
Comment (barnjamin) [reply go_templates/query.vm:43]: I don't quite understand

The request generated is still a POST method but the URL needs to include the parameter.

**Review by michaeldiamant — APPROVED** [review]
As of https://github.com/algorand/generator/pull/37/commits/c5fb80f83dcb14c682e2cdf02be6510e14599e2d and as expected, I confirmed re-generating Go SDK produces no diffs in https://github.com/algorand/go-algorand-sdk/pull/335.

**Comment by michaeldiamant** [comment]
Merging - Based on a verbal with @winder, we agreed I can merge provided I'm comfortable with the diff.

**Pull request merged**